### PR TITLE
Pwa 559 sg analytics

### PR DESCRIPTION
**Pull request by Carsten89**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -62,6 +62,7 @@ typings/
 .vscode
 node_modules
 dist
+frontend/package-lock.json
```
Comment (kernelkiller) [inline .gitignore]: You could additionally add a `.npmrc` with `package-lock=false`

File: frontend/src/Plugin.js [hunk 2]
```diff
@@ -23,21 +15,49 @@ class ShopgateAnalytics extends SgTrackingPlugin {
...
 
     initSDK(stage);
+    this.config = {};
 
-    sgAnalytics('setConfig', {
+    this.updateConfig({
       shopNumber,
       pushToken,
-      sgUserId: userId,
+      sgUserId: userId ? `${userId}` : userId,
```
Comment (kernelkiller) [inline frontend/src/Plugin.js]: Does tracking with the `userId` comply with the GDPR? And is it already prepared to delete the tracking data when the user requests it?

File: frontend/src/index.js [hunk 3]
```diff
@@ -14,8 +10,24 @@ import config from '../config';
 * @return {Object}
 */
 export default function init(options) {
-  return new Plugin({
-    ...config,
+  const { state, subscribe } = options;
```
Comment (kernelkiller) [inline frontend/src/index.js]: I'd say it's better to register a subscription via the `extension-config` to enable listening to the `userDataReceived$`. You could use a common event to update the specific plugin instance.

File: frontend/src/index.js [hunk 4]
```diff
@@ -1,10 +1,6 @@
-/**
- * Copyright (c) 2017, Shopgate, Inc. All rights reserved.
- *
- * This source code is licensed under the Apache 2.0 license found in the
- * LICENSE file in the root directory of this source tree.
-*/
-
+import { shopNumber } from '@shopgate/pwa-common/helpers/config';
+import { getUserData } from '@shopgate/pwa-common/selectors/user';
+import { userDataReceived$ } from '@shopgate/pwa-common/streams/user';
```
Comment (kernelkiller) [inline frontend/src/index.js]: The `pwa-common` module isn't available within the `package.json`

File: frontend/src/Plugin.js (hunk 2 above)
Comment (Carsten89) [reply frontend/src/Plugin.js]: i contacted our POs

File: frontend/src/index.js (hunk 3 above)
Comment (Carsten89) [reply frontend/src/index.js]: done

File: frontend/src/index.js (hunk 4 above)
Comment (Carsten89) [reply frontend/src/index.js]: done

**Pull request merged**